exchange/bidder.go

```diff
@@ -58,17 +58,32 @@ type AdaptedBidder interface {
 	//
 	// Any errors will be user-facing in the API.
 	// Error messages should help publishers understand what might account for "bad" bids.
-	requestBid(ctx context.Context, bidderRequest BidderRequest, conversions currency.Conversions, reqInfo *adapters.ExtraRequestInfo, adsCertSigner adscert.Signer, bidRequestOptions bidRequestOptions, alternateBidderCodes openrtb_ext.ExtAlternateBidderCodes, hookExecutor hookexecution.StageExecutor, ruleToAdjustments openrtb_ext.AdjustmentsByDealID) ([]*entities.PbsOrtbSeatBid, []error)
+	requestBid(ctx context.Context, bidderRequest BidderRequest, conversions currency.Conversions, reqInfo *adapters.ExtraRequestInfo, adsCertSigner adscert.Signer, requestBidInfo extraRequestBidInfo, alternateBidderCodes openrtb_ext.ExtAlternateBidderCodes, hookExecutor hookexecution.StageExecutor, ruleToAdjustments openrtb_ext.AdjustmentsByDealID) ([]*entities.PbsOrtbSeatBid, []error)
 }
 
-// bidRequestOptions holds additional options for bid request execution to maintain clean code and reasonable number of parameters
-type bidRequestOptions struct {
+type extraRequestBidInfo struct {
```

Comment:
@onkarvhanumante Since we're no longer trying to get each go routine update a common time field, do you think we still need this struct here that encapsulates request and response info? 

What do you think about having `requestBid` return `bidderResponseInfo` (that wraps the `bidderResponseStartTime` value and can hold additional response related information in the future if needed) and `getAllBids` can read from that and set the `bidderResponseStartTime` value in `BidResponseWrapper`. `GetAllBids` will still return `extraAuctionResponseInfo` like it's doing currently in this PR.

To give you a better idea, I tried these changes on top of master and pushed it [here](https://github.com/prebid/prebid-server/compare/master...mansinahar:prebid-server:response-preparation-metric?expand=1). Please note that this is just to give you a reference of what I mean and I haven't verified these changes to be complete so you may or may not need to make more changes for them to work.

@VeronikaSolovei9 @guscarreon @hhhjort It will be nice to know what you think as well as you all are the official reviewers on this PR.

Reply:
I don't oppose this. It'll be a pain to add a third return value to `requestBid()` everywhere, but I'm OK either way.

Reply:
Yeah, I agree. It's definitely gonna be a pain to update it everywhere. However, I do think it will be useful to add it as it also lets us support more such response related fields in the future, if needed. For instance, the non-bid PR that's open right now adds another return parameter to `requestBid` that _may_ be a potential candidate for this struct.

Reply:
Yes, this is a good point. Basically `requestBid` will receive `extraRequestBidInfo` and return `bidderResponseInfo`.
This will also remove names confusion like `requestBidInfo.response.bidderResponseStartTime`